File: Assets/zfoolua/common/PairIntLong.lua

local PairIntLong = {}

function PairIntLong:new()
    local obj = {
        key = 0, -- int
        value = 0 -- long
    }
    setmetatable(obj, self)
    self.__index = self
    return obj
end

function PairIntLong:protocolId()
    return 110
end

function PairIntLong:write(buffer, packet)
    if packet == nil then
        buffer:writeInt(0)
        return
    end
    buffer:writeInt(-1)
    buffer:writeInt(packet.key)
    buffer:writeLong(packet.value)
end

function PairIntLong:read(buffer)
    local length = buffer:readInt()
    if length == 0 then
        return nil
    end
    local beforeReadIndex = buffer:getReadOffset()
    local packet = PairIntLong:new()
    local result0 = buffer:readInt()
    packet.key = result0
    local result1 = buffer:readLong()
    packet.value = result1
    if length > 0 then
        buffer:setReadOffset(beforeReadIndex + length)
    end
    return packet
end

return PairIntLong

File: Assets/zfoocs/Common/PairIntLong.cs
using System;
using System.Collections.Generic;
namespace zfoocs
{
    
    public class PairIntLong
    {
        public int key;
        public long value;
    }

    public class PairIntLongRegistration : IProtocolRegistration
    {
        public short ProtocolId()
        {
            return 110;
        }
    
        public void Write(ByteBuffer buffer, object packet)
        {
            if (packet == null)
            {
                buffer.WriteInt(0);
                return;
            }
            PairIntLong message = (PairIntLong) packet;
            buffer.WriteInt(-1);
            buffer.WriteInt(message.key);
            buffer.WriteLong(message.value);
        }
    
        public object Read(ByteBuffer buffer)
        {
            int length = buffer.ReadInt();
            if (length == 0)
            {
                return null;
            }
            int beforeReadIndex = buffer.GetReadOffset();
            PairIntLong packet = new PairIntLong();
            int result0 = buffer.ReadInt();
            packet.key = result0;
            long result1 = buffer.ReadLong();
            packet.value = result1;
            if (length > 0)
            {
                buffer.SetReadOffset(beforeReadIndex + length);
            }
            return packet;
        }
    }
}